tensorboard/compat/tensorflow_stub/io/gfile.py

```diff
@@ -228,6 +248,22 @@ def read(self, filename, binary_mode=False, size=None, offset=None):
         else:
             return stream.decode('utf-8')
 
+    def write(self, filename, file_content, binary_mode=False):
+        """Writes string file contents to a file.
+
+        Args:
+            filename: string, a path
+            file_content: string, the contents
+            binary_mode: bool, write as binary if True, otherwise text
+        """
+        client = boto3.client("s3")
+        bucket, path = self.bucket_and_path(filename)
+        if binary_mode:
```

Comment:
What's the behavior of the boto3 APIs if you pass text vs binary?  Does the text get encoded in some particular encoding?  If so, can we specify it (UTF-8) so it's explicit?

Reply:
UTF-8 is the standard and I think everything gets stored as that in the default case.

Reply:
Unresolving since I think this is still an issue.  I think the fix is just to change the if statement here so that if we're if we're in binary mode, we just check that we got a `six.binary_type` for `file_content` and raise a TypeError if we didn't, else if we're not in binary mode, we call `compat.as_bytes(file_content)` to convert the text string to a binary string.

That way we ensure we're always passing a binary string to put_object().